chirpr.py

```diff
@@ -119,9 +128,37 @@ def chirps():
     return render_template('admin/chirps.html', chirps=chirp_list)
     
     
+@app.route('/chirps/<handle>')
+def chirps_hn(handle):
+    uid = db_access.get_id(handle)
+    return redirect('%s?filter=%s'%(url_for('chirp'), handle))
+
+    
+@app.route('/chirp', methods=['GET', 'POST'])
+def chirp():
+    if 'user' in session:
+        user = session['user']
+        MD = False
+        if request.method == 'POST':
+            content = request.form["content"]
+            if 'MD' in request.form:
+                if request.form['MD'] == 'on':
+                    content = markdown2.markdown(content)
+                    MD = True
+            db_access.add_chirp(content,user,MD=MD)
```

Comment:
I don't like you putting markup in the database. You have some options here. You could add a column that indicates that this is a chirp with markdown. Or you could assume all chirps have markdown.

But either way, the DB should contain the raw data, and should not be concerned with presentation. The `markdown` version should be only used when you are presenting the data to the user.